omod/src/main/java/org/openmrs/module/owa/web/controller/SettingsFormController.java

```diff
@@ -20,72 +16,79 @@
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.context.request.WebRequest;
 
+import javax.servlet.http.HttpServletRequest;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
 @Controller
 @RequestMapping("/module/owa/settings")
 public class SettingsFormController {
-	
+
 	private static final Log log = LogFactory.getLog(SettingsFormController.class);
-	
+
 	@RequestMapping(method = RequestMethod.GET)
 	public void showForm() {
 	}
-	
+
 	@RequestMapping(method = RequestMethod.POST)
 	public String handleSubmission(@ModelAttribute("globalPropertiesModel") GlobalPropertiesModel globalPropertiesModel,
-	        Errors errors, WebRequest request, HttpServletRequest req) {
+			Errors errors, WebRequest request, HttpServletRequest req) {
 		if (Context.hasPrivilege("Manage OWA")) {
 			globalPropertiesModel.validate(globalPropertiesModel, errors);
 			if (errors.hasErrors())
 				return null; // show the form again
-				
+
 			AdministrationService administrationService = Context.getAdministrationService();
 			for (GlobalProperty p : globalPropertiesModel.getProperties()) {
 				if (p.getProperty().equals(AppManager.KEY_APP_FOLDER_PATH) && p.getPropertyValue().equals("")) {
-					p.setPropertyValue(OpenmrsUtil.getApplicationDataDirectory() + "owa");
+					p.setPropertyValue(OpenmrsUtil.getApplicationDataDirectory()
+							+ (OpenmrsUtil.getApplicationDataDirectory().endsWith(File.separator) ? "owa" : File.separator
```

Comment:
It should be 
OpenmrsUtil.getApplicationDataDirectory().endsWith(File.separator) ? "owa" : File.separator + "owa"